java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -725,7 +729,19 @@ private void executeVariable(VariableTree variableTree, @Nullable Tree terminato
       } else if (variableTree.type().symbolType().is("boolean")) {
         sv = SymbolicValue.FALSE_LITERAL;
       } else if (variableTree.parent().is(Tree.Kind.CATCH)) {
-        sv = constraintManager.createSymbolicValue(variableTree);
+        Type caughtType = variableTree.symbol().type();
+        Type exceptionType = null;
+        if (programState.peekValue() instanceof SymbolicValue.ExceptionalSymbolicValue) {
```

Comment:
This if is actually not really required and more a revelation of some issue where we lack a correct stacking of SV. Please at the very least add a comment regarding this and create a ticket to handle cases where this condition is not satisfied.